api/grpc/Server.go

```diff
@@ -99,9 +115,13 @@ func (s *Server) GetName() string {
 
 // CheckPermission processes an authorization check and returns whether or not the operation would be allowed
 func (s *Server) CheckPermission(ctx context.Context, rpcReq *core.CheckPermissionRequest) (*core.CheckPermissionResponse, error) {
+	requestor, err := s.getRequestorIdentityFromGrpcContext(ctx)
+	if err != nil {
+		return nil, err
+	}
 
 	req := application.CheckRequest{
-		Requestor:    getRequestorIdentityFromGrpcContext(ctx),
+		Requestor:    requestor,
```

Comment:
As said above, I think this could be a string at this point, to be looked up for a real user in the app services/repository layer.

Reply:
It could. It doesn't need to convert all the way to a Principal necessarily, but it _is_ meant to have a placeholder there to resolve an opaque token to an identity that's just sort of a passthrough atm. Making that resolution into its own thing and separating the resolve from the lookup could make a lot of sense, maybe like the previous comment but happening at different layers and letting the requestor be a string.

That could make capturing more info from token introspection tricky (ie: it's opaque where the data comes from right now, but if we separate it into getting the id from one source and getting the whole user from another, that doesn't let us, say, grab all the user data locally if available in the token), but maybe that doesn't matter. We'll most likely be getting it from user service anyway.